### PR TITLE
Fix RTSP and ffmpeg source adapters

### LINKED ISSUE #471: OpenCV CUDA BGRemover Benchmark results in buggy behavior
Error log is:

```
samples/opencv_cuda_bg_remover_mog2/run_perf.sh
 INFO  insight::savant::config::module_config > Configure module...
 WARN  insight::savant::config::json_resolver > JSON loads fail, returning None for "None".
 WARN  insight::savant::config::json_resolver > JSON loads fail, returning None for "None".
 WARN  insight::savant::config::json_resolver > JSON loads fail, returning None for "None".
 WARN  insight::savant::config::json_resolver > JSON loads fail, returning None for "None".
 INFO  insight::savant::config::module_config > Configure pipeline elements...
 INFO  insight::savant::config::module_config > Pipeline batch size is set to 1.
 INFO  insight::savant::config::module_config > Module configuration is complete.

(gst-plugin-scanner:19): GStreamer-WARNING **: 06:39:02.305: Failed to load plugin '/usr/lib/x86_64-linux-gnu/gstreamer-1.0/libgstmpeg2enc.so': libmpeg2encpp-2.1.so.0: cannot open shared object file: No such file or directory

(gst-plugin-scanner:19): GStreamer-WARNING **: 06:39:02.404: Failed to load plugin '/usr/lib/x86_64-linux-gnu/gstreamer-1.0/libgstmpeg2dec.so': libmpeg2.so.0: cannot open shared object file: No such file or directory

(gst-plugin-scanner:19): GStreamer-WARNING **: 06:39:03.581: Failed to load plugin '/usr/lib/x86_64-linux-gnu/gstreamer-1.0/deepstream/libnvdsgst_ucx.so': libucs.so.0: cannot open shared object file: No such file or directory

(gst-plugin-scanner:19): GStreamer-WARNING **: 06:39:03.616: Failed to load plugin '/usr/lib/x86_64-linux-gnu/gstreamer-1.0/deepstream/libnvdsgst_inferserver.so': libtritonserver.so: cannot open shared object file: No such file or directory

(gst-plugin-scanner:19): GStreamer-WARNING **: 06:39:03.622: Failed to load plugin '/usr/lib/x86_64-linux-gnu/gstreamer-1.0/deepstream/libnvdsgst_udp.so': librivermax.so.0: cannot open shared object file: No such file or directory
 INFO  insight::savant::deepstream::encoding  > Checking if encoder for codec 'hevc' is available
 INFO  insight::savant::gstreamer::runner     > Starting pipeline `<Gst.Pipeline object at 0x7f02e2587600 (GstPipeline at 0x30ac140)>`...
 INFO  insight::savant::gstreamer::runner     > The pipeline is initialized and ready to process data. Initialization took 0:00:00.001684.
 INFO  insight::savant::gstreamer::runner     > Received EOS from pipeline0.
 INFO  insight::savant::gstreamer::runner     > The pipeline is about to stop. Operation took 0:00:00.238575.
 INFO  insight::savant::deepstream::encoding  > Encoder for codec 'hevc' is available
 INFO  insight::savant::deepstream::utils::pipeline > No telemetry provider specified. Using noop tracer.
 INFO  insight::savant::demo                        > Pipeline frame processing parameters: {'width': 2560, 'height': 720, 'batch-size': 1, 'buffer-pool-size': 4, 'batched-push-timeout': 2000, 'live-source': False, 'interpolation-method': 6, 'drop-pipeline-eos': False, 'nvbuf-memory-type': 3}.
 INFO  insight::savant::gstreamer::runner           > Starting pipeline `demo<NvDsPipeline>: uridecodebin:v1(name=source) -> nvstreammux:v1(name=muxer) -> pyfunc:v1(name=pyfunc+gstpluginpyfunc0) -> nvstreamdemux:v1(name=demuxer)`...
 INFO  insight::savant::healthcheck::status         > Setting module status to ModuleStatus.STARTING.
 INFO  insight::savant::gstreamer::runner           > The pipeline is initialized and ready to process data. Initialization took 0:00:00.013784.
 INFO  insight::savant::healthcheck::status         > Setting module status to ModuleStatus.RUNNING.
 WARN  insight::savant::gstreamer::runner           > Received warning gst-stream-error-quark: No decoder available for type 'audio/mpeg, mpegversion=(int)4, framed=(boolean)true, stream-format=(string)raw, level=(string)2, base-profile=(string)lc, profile=(string)lc, codec_data=(buffer)121056e500, rate=(int)44100, channels=(int)2'. (6). gsturidecodebin.c(920): unknown_type_cb (): /GstPipeline:demo/GstURIDecodeBin:source
 INFO  insight::savant::demo                        > Added source 0
 INFO  insight::savant::savant_rs_add_frames        > Sink caps changed to video/x-raw(memory:NVMM), format=(string)NV12, width=(int)1280, height=(int)720, interlace-mode=(string)progressive, multiview-mode=(string)mono, multiview-flags=(GstVideoMultiviewFlagsSet)0:ffffffff:/right-view-first/left-flipped/left-flopped/right-flipped/right-flopped/half-aspect/mixed-mono, pixel-aspect-ratio=(fraction)1/1, chroma-site=(string)mpeg2, colorimetry=(string)bt709, framerate=(fraction)30/1, nvbuf-memory-type=(string)nvbuf-mem-cuda-device, gpu-id=(int)0
nvstreammux: Successfully handled EOS for source_id=0
thread '<unnamed>' panicked at 'Failed to move objects to sink-capsfilter, error: Object 18364 location not found.', savant_python/src/capi/pipeline.rs:23:29
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
fatal runtime error: failed to initiate panic, error 5
```

### PR DESCRIPTION
**Pull request by tomskikh**

### CODE REVIEW

File: adapters/gst/gst_plugins/python/savant_parse_bin.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+            {Gst.EventType.CAPS: self.on_sink_pad_caps},
+        )
+
+    def on_sink_pad_caps(self, pad: Gst.Pad, event: Gst.Event):
+        caps: Gst.Caps = event.parse_caps()
+        self.logger.debug('Got caps %r. Trying to find parser.', caps.to_string())
+        try:
+            codec = CODEC_BY_CAPS_NAME[caps[0].get_name()]
+        except KeyError:
+            error = f'Cannot found parser for caps {caps.to_string()!r}.'
```
Comment (denisvmedyantsev) [inline adapters/gst/gst_plugins/python/savant_parse_bin.py]: Is it not enough to set the correct caps for sink/source templates? 

L14 `Gst.Caps.from_string(';'.join(x.value.caps_name for x in Codec))`

File: libs/gstsavantframemeta/gstsavantframemeta/src/gstsavantbatchmeta.cpp [hunk 2]
```diff
@@ -82,21 +82,12 @@ GstSavantBatchMeta *gst_buffer_add_savant_batch_meta(GstBuffer *buffer,
                            idx);
         return NULL;
     }
-    meta = gst_buffer_get_savant_batch_meta(buffer);
```
Comment (denisvmedyantsev) [inline libs/gstsavantframemeta/gstsavantframemeta/src/gstsavantbatchmeta.cpp:85]: Could you please explain the reason for this change?

Why did we previously need to check if meta was added, but now we don’t?

**Review by denisvmedyantsev — APPROVED** [review]

File: adapters/gst/gst_plugins/python/savant_parse_bin.py (hunk 1 above)
Comment (tomskikh) [reply adapters/gst/gst_plugins/python/savant_parse_bin.py]: Probably that should be enough. I'll check it.

File: libs/gstsavantframemeta/gstsavantframemeta/src/gstsavantbatchmeta.cpp (hunk 2 above)
Comment (tomskikh) [reply libs/gstsavantframemeta/gstsavantframemeta/src/gstsavantbatchmeta.cpp:85]: There was a bug when using RTSP cameras with h265 codec: https://github.com/insight-platform/Savant/issues/468. H265 parser or decoder on the module was reusing buffers that already have meta. Checking if the buffer already has meta fixed the issue but leaded to the issue with benchmarks (`opencv_cuda_bg_remover_mog2` and some others). In those benchmarks some elements were also reusing buffers but they're doing it correctly. It looks like a bug in h265parse: it doesn't reuse buffers correctly in some cases. So I've reverted the fix and fixed the issue by adding a parser in RTSP source adapter.

File: adapters/gst/gst_plugins/python/savant_parse_bin.py (hunk 1 above)
Comment (tomskikh) [reply adapters/gst/gst_plugins/python/savant_parse_bin.py]: Yes, it's enough to set caps for sink. Removed check.

**Pull request merged**